src/index.js

```diff
@@ -0,0 +1,135 @@
+      renderTasks();
+    });
+
+    listItem.querySelector('.editable').addEventListener('click', (event) => {
+      event.target.contentEditable = true;
+      event.target.focus();
+      setCaretToEnd(event.target);
+      event.stopPropagation();
+    });
+
+    listItem.addEventListener('click', (event) => {
+      const trashIcon = listItem.querySelector('.fa-trash-alt');
+      const isSelected = listItem.classList.contains('selected');
+
+      if (event.target !== trashIcon) {
+        if (!isSelected) {
+          listItem.classList.add('selected');
+          trashIcon.style.display = 'inline';
+          listItem.querySelector('.fa-ellipsis-v').style.display = 'none';
+        } else {
+          listItem.classList.remove('selected');
+          trashIcon.style.display = 'none';
+          listItem.querySelector('.fa-ellipsis-v').style.display = 'inline';
+        }
+      }
+    });
+
+    listItem.querySelector('.fa-trash-alt').addEventListener('click', (event) => {
+      event.stopPropagation();
+      tasks = deleteTask(tasks, index);
+      renderTasks();
+    });
+
+    listItem.querySelector('.editable').addEventListener('blur', function editableBlur() {
+      this.contentEditable = false;
+      tasks = editTask(tasks, index, this.textContent.trim());
+    });
+
+    if (task.completed) {
+      listItem.querySelector('.editable').style.textDecoration = 'line-through';
+    }
+
+    taskList.appendChild(listItem);
+  });
+};
+
+const taskInput = document.getElementById('task-input');
+const taskForm = document.getElementById('task-form');
+
+const handleAddTask = (event) => {
+  event.preventDefault();
+  const taskDescription = taskInput.value.trim();
+  if (taskDescription !== '') {
+    tasks = addTask(tasks, taskDescription);
+    taskInput.value = '';
+    renderTasks();
+    saveTasks(tasks);
+  }
+};
+
+taskForm.addEventListener('submit', handleAddTask);
+
+const enterIcon = document.querySelector('.icon-container');
+
+enterIcon.addEventListener('click', (event) => {
+  event.preventDefault();
+  const taskDescription = taskInput.value.trim();
+  if (taskDescription !== '') {
+    tasks = addTask(tasks, taskDescription);
+    taskInput.value = '';
+    renderTasks();
+    saveTasks(tasks);
+  }
+});
+
+enterIcon.addEventListener('keypress', (event) => {
+  if (event.key === 'Enter') {
+    event.preventDefault();
+    const taskDescription = taskInput.value.trim();
+    if (taskDescription !== '') {
+      tasks = addTask(tasks, taskDescription);
+      taskInput.value = '';
+      renderTasks();
+      saveTasks(tasks);
+    }
+  }
+});
+
+renderTasks();
+saveTasks(tasks);
+
+const clearButton = document.getElementById('clear-completed');
+
+clearButton.addEventListener('click', () => {
+  tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+  tasks = tasks.filter((task) => !task.completed);
+
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+
+  renderTasks();
+});
```

Comment:
- [ ] Let's build our apps step-by-step as we're required to. The clear-completed feature is not part of this milestone and is not required to implement a feature that should be done in future milestones. Therefore, just as highlighted by the previous reviewer, kindly remove these codes as it's not part of this milestone for adding, deleting, and updating tasks. It's a feature you will be working on in a future milestone coming up next. 👍🏽